presto-main/src/main/java/com/facebook/presto/cost/ScalarStatsAnnotationProcessor.java

```diff
@@ -0,0 +1,236 @@
+                    sourceStats.stream().map(VariableStatsEstimate::getNullsFraction).collect(toImmutableList()),
+                    paramIndexVsStatsMap.getKey(),
+                    applyPropagateAllStats(propagateAllStats, scalarPropagateSourceStats.nullFraction()))), 1.0);
+            distinctValuesCount = firstFiniteValue(distinctValuesCount, processSingleArgumentStatistic(outputRowCount, nullFraction, callExpression,
+                    sourceStats.stream().map(VariableStatsEstimate::getDistinctValuesCount).collect(toImmutableList()),
+                    paramIndexVsStatsMap.getKey(),
+                    applyPropagateAllStats(propagateAllStats, scalarPropagateSourceStats.distinctValuesCount())));
+            StatsPropagationBehavior averageRowSizeStatsBehaviour = applyPropagateAllStats(propagateAllStats, scalarPropagateSourceStats.avgRowSize());
+            averageRowSize = minExcludingNaNs(firstFiniteValue(averageRowSize, processSingleArgumentStatistic(outputRowCount, nullFraction, callExpression,
+                    sourceStats.stream().map(VariableStatsEstimate::getAverageRowSize).collect(toImmutableList()),
+                    paramIndexVsStatsMap.getKey(),
+                    averageRowSizeStatsBehaviour)), returnNaNIfTypeWidthUnknown(getReturnTypeWidth(callExpression, averageRowSizeStatsBehaviour)));
+            maxValue = firstFiniteValue(maxValue, processSingleArgumentStatistic(outputRowCount, nullFraction, callExpression,
+                    sourceStats.stream().map(VariableStatsEstimate::getHighValue).collect(toImmutableList()),
+                    paramIndexVsStatsMap.getKey(),
+                    applyPropagateAllStats(propagateAllStats, scalarPropagateSourceStats.maxValue())));
+            minValue = firstFiniteValue(minValue, processSingleArgumentStatistic(outputRowCount, nullFraction, callExpression,
+                    sourceStats.stream().map(VariableStatsEstimate::getLowValue).collect(toImmutableList()),
+                    paramIndexVsStatsMap.getKey(),
+                    applyPropagateAllStats(propagateAllStats, scalarPropagateSourceStats.minValue())));
+        }
+        if (isNaN(maxValue) || isNaN(minValue)) {
+            minValue = NaN;
+            maxValue = NaN;
+        }
+        return VariableStatsEstimate.builder()
+                .setLowValue(minValue)
+                .setHighValue(maxValue)
+                .setNullsFraction(nullFraction)
+                .setAverageRowSize(firstFiniteValue(scalarStatsHeader.getAvgRowSize(), averageRowSize, returnNaNIfTypeWidthUnknown(getReturnTypeWidth(callExpression, UNKNOWN))))
+                .setDistinctValuesCount(processDistinctValuesCount(outputRowCount, nullFraction, scalarStatsHeader.getDistinctValuesCount(), distinctValuesCount)).build();
+    }
+
+    private static double processDistinctValuesCount(double outputRowCount, double nullFraction, double distinctValuesCountFromConstant, double distinctValuesCount)
```

Comment:
The counts should ints

Reply:
These and others are double values across the code base, I am not sure if we can change that.

Reply:
I'm pretty sure we can. It might be inconvenient to do in this PR, though I'm not sure about that. New methods absolutely can use ints/longs for things that are integers. Just maybe you need to pass in a value from existing code that is already *incorrectly* a double,. If so, and if we need to move forward with this before fixing that, then please file a detailed issue. And also consider using the right type in the new methods and casting the value before passing it into the new methods so the source of the bad type is clear. 

However, very likely many if not all of these new arguments and variables can simply be made ints now.

Reply:
The https://github.com/prestodb/presto/pull/23545#discussion_r1793207848 specifies the reason for using doubles here and other places.